operator/src/main/java/oracle/kubernetes/operator/DomainProcessorImpl.java

```diff
@@ -974,42 +980,86 @@ public NextAction apply(Packet packet) {
   }
 
   private static class DomainStatusUpdate {
-    private final Domain domain;
     private final V1Pod pod;
     private final String domainUid;
+    private DomainProcessorDelegate delegate = null;
+    private DomainPresenceInfo info = null;
+    private PodWatcher.PodStatus podStatus;
 
-    DomainStatusUpdate(Domain domain, V1Pod pod, String domainUid) {
-      this.domain = domain;
+    DomainStatusUpdate(V1Pod pod, String domainUid, DomainProcessorDelegate delegate,
+                       DomainPresenceInfo info, PodWatcher.PodStatus podStatus) {
       this.pod = pod;
       this.domainUid = domainUid;
+      this.delegate = delegate;
+      this.info = info;
+      this.podStatus = podStatus;
     }
 
-    public void invoke() {
-      Optional.ofNullable(getMatchingContainerStatus())
-            .map(V1ContainerStatus::getState)
-            .map(V1ContainerState::getWaiting)
-            .ifPresent(waiting -> updateStatus(waiting.getReason(), waiting.getMessage()));
-    }
-
-    private void updateStatus(String reason, String message) {
-      if (reason == null || message == null) {
-        return;
+    private void invoke() {
+      switch (podStatus) {
+        case PHASE_FAILED:
+          DomainStatusUpdater.createFailedStep(
```

Comment:
This branch creates the FailedStep. Where is it being run?

Reply:
Thanks for catching this, I meant to run it with "delegate.runSteps" and missed it. I will change it like this =>
          delegate.runSteps(
                  DomainStatusUpdater.createFailedStep(
                          info, pod.getStatus().getReason(), pod.getStatus().getMessage(), null));

Reply:
That suggests that you are missing a unit test. If you had the test, it would have caught it.

Reply:
Added unit test IntrospectionStatusTest.whenIntrospectorJobPodPhaseFailed_patchDomain() to cover this. Thanks.